src/library/connMatrixTools.cc

```diff
@@ -98,11 +102,33 @@ namespace constel {
   */
   void fillconnMatrixNoSmoothing(
       Connectivities *conn_ptr, QuickMap &fiberExtremity1NeighMeshVertex,
-      QuickMap & fiberExtremity2NeighMeshVertex) {
+      QuickMap & fiberExtremity2NeighMeshVertex)
+  {
     Connectivities &conn = *conn_ptr;
     double isz
       = 1. / (fiberExtremity1NeighMeshVertex.size()
               * fiberExtremity2NeighMeshVertex.size());
+    for (QuickMap::const_iterator neigh_2
+           = fiberExtremity2NeighMeshVertex.begin();
+         neigh_2 != fiberExtremity2NeighMeshVertex.end(); ++neigh_2)
+    {
+      for (QuickMap::const_iterator neigh_1
+             = fiberExtremity1NeighMeshVertex.begin();
+           neigh_1 != fiberExtremity1NeighMeshVertex.end(); ++neigh_1)
+      {
+        conn[neigh_1->first][neigh_2->first] += isz;
+        conn[neigh_2->first][neigh_1->first] += isz;
+      }
+    }
+  }
+
+  void fillWeightedConnMatrixNoSmoothing(
```

Comment:
Why making the weighted fibers a different case, and not just assuming all fibers are weighted, with a default weight to 1 when no weight file is given ?